x-pack/winlogbeat/module/security/config/winlogbeat-security.js

```diff
@@ -1495,12 +1739,118 @@ var security = (function () {
+            aceResult['type'] = aceTypes[aceType];
+        }
+        if (acePerm) {
+            if (/^0x/.test(acePerm)) {
+                var perms = translatePermissionMask(acePerm);
+            }
+            else {
+                var perms = []
+                var permPairs = acePerm.match(/.{1,2}/g);
+                for ( var i = 0; i < permPairs.length; i ++) {
+                    perms.push(permissionDescription[permPairs[i]])
+                }
+            }
+            aceResult['perms'] = perms;
+        }
+        return aceResult;
+    };
+
+    var enrichSDDL = function(evt, sddl) {
+        var sddlStr = evt.Get(sddl);
+        if (!sddlStr) {
+            return;
+        }
+        var sdOwner = sddlStr.match(/^O\:[A-Z]{2}/g);
+        var sdGroup = sddlStr.match(/^G\:[A-Z]{2}/g);
+        var sdDacl = sddlStr.match(/(D:([A-Z]*(\(.*\))*))/g);
+        var sdSacl = sddlStr.match(/(S:([A-Z]*(\(.*\))*))?$/g);
+        if (sdOwner) {
+            evt.Put(sddl+"Owner", translateSID(sdOwner));
+        }
+        if (sdGroup) {
+            evt.Put(sddl+"Group", translateSID(sdGroup));
+        }
+        if (sdDacl) {
+            // Split each entry of the DACL
+            var daclList = (sdDacl[0]).match(/\([^*\)]*\)/g);
+            if (daclList) {
+                for (var i = 0; i < daclList.length; i++) {
+                    var newDacl = translateACL(daclList[i].replace("(", '').replace(")", ''));
+                    evt.Put(sddl+"Dacl"+i, newDacl['grantee']+" :"+newDacl['type']+" ("+newDacl['perms']+")");
+                    if ( newDacl['grantee'] === "Administrator" || newDacl['grantee'] === "Guest" || newDacl['grantee'] === "KRBTGT" ) {
+                        evt.AppendTo('related.user', newDacl['grantee']);
+                    }
+                }
+            }
+        }
+        if (sdSacl) {
+            // Split each entry of the SACL
+            var saclList = (sdSacl[0]).match(/\([^*\)]*\)/g);
+            if (saclList) {
+                for (var i = 0; i < saclList.length; i++) {
+                    var newSacl = translateACL(saclList[i].replace("(", '').replace(")", ''));
+                    evt.Put(sddl+"Sacl"+i, newSacl['grantee']+" :"+newSacl['type']+" ("+newSacl['perms']+")");
+                    if ( newSacl['grantee'] === "Administrator" || newSacl['grantee'] === "Guest" || newSacl['grantee'] === "KRBTGT" ) {
+                        evt.AppendTo('related.user', newSacl['grantee']);
+                    }
+                }
+            }
+        }
+    };
+
     var addSessionData = new processor.Chain()
         .Convert({
             fields: [
                 {from: "winlog.event_data.AccountName", to: "user.name"},
                 {from: "winlog.event_data.AccountDomain", to: "user.domain"},
                 {from: "winlog.event_data.ClientAddress", to: "source.ip", type: "ip"},
+                {from: "winlog.event_data.ClientAddress", to: "related.ip"},
```

Comment:
I think we want to add `type: "ip"`, I know we had at least one bug with this for with `source.ip`